gkr_iop/src/chip.rs

```diff
@@ -0,0 +1,28 @@
+use crate::{evaluation::EvalExpression, gkr::layer::Layer};
+
+pub mod builder;
+pub mod protocol;
+
+/// Chip stores all information required in the GKR protocol, including the 
+/// commit phases, the GKR phase and the opening phase.
+#[derive(Clone, Debug, Default)]
+pub struct Chip {
```

Comment:
We can separate ChipBuilder vs ConstrainSystem.
Refer similar design in ceno_zkvm https://github.com/scroll-tech/ceno/blob/master/ceno_zkvm/src/circuit_builder.rs#L487-L488
All the information store in this chip struct should belong to `ConstrainSystem`.
So it can be Serializable.

similar as commment here https://github.com/scroll-tech/ceno/pull/799/files#r1904053980